tools/azsdk-cli/docs/specs/3-customizing-client-tsp.spec.md

```diff
@@ -0,0 +1,116 @@
+
+### Current State
+
+Currently, feedback for adding client customizations to `client.tsp` can come in a few forms:
+
+1. From reviews, e.g. API View feedback
+2. From automated processes, e.g. code analyzers
+3. From manually triggered processes, e.g. breaking changes analysis
+
+**.NET**:
+The .NET SDK has code analyzers that run during their SDK build (`dotnet build`) that identifies problematic code (e.g. class names that don't follow .NET naming conventions) and suggests fixes.
+Many of these fixes are handled by adding customizations to `client.tsp` for .NET, such as renaming models with `@clientName`. Currently these fixes are manually applied after SDK builds fail.
+_.NET has a clear flow for identifying and applying `client.tsp` customizations via: Emit SDK -> Build SDK -> Apply fixes -> Repeat._
+
+**Go**:
+The Go SDK has a [breaking changes](https://github.com/Azure/azure-sdk-for-go/blob/instruction/.github/prompts/go-sdk-breaking-changes-review.prompts.md) prompt that analyzes the generated SDK's changelog for specific patterns and maps those to `client.tsp` customizations where possible.
+_We don't currently have a clear step in our process for identifying breaking changes, but the flow for Go is: Emit SDK -> Analyze changelog & apply fixes -> Repeat._
+
+**All Languages**:
+All language SDKs have their APIs reviewed in API View. Common feedback includes renaming models per language, or changing operations that appear in a client. We have documentation for customizing SDKs such as [renaming types](https://azure.github.io/typespec-azure/docs/howtos/generate-client-libraries/09renaming/), but recent (10/14/2025) testing shows that even with the current documentation, AI agents have difficulty applying these customizations correctly in a consistent manner: particularly when `scope` is required.
+
+There has already been some work done to support identifying and fixing code customizations - customizations made directly to generated SDKs rather than to the TypeSpec files - in the `TspClientUpdateTool`. This tool exists as an MCP tool/CLI command and currently works _after_ an SDK has been generated from TypeSpec.
+
+#### Gaps
+
+The current state exposes some gaps we have today:
+
+- **AI agents are not experts on client customizations.** They make mistakes when applying customizations to `client.tsp`, even with access to the existing documentation.
+- **AI agents don't know when to suggest client customizations.** There is no mechanism in azsdk-cli today to steer AI agents to focus on applying `client.tsp` customizations.
+- **Nothing shared across languages.** Each language team has their own process for identifying and applying `client.tsp` customizations.
+- **Difficult to apply changes from outside the inner loop.** We lack a process for updating `client.tsp` from outside the azure-rest-api-specs repo.
```

Comment:
+1 to this, it's been awkward in .NET trying to make adjustments without a local copy of the specs repo.

Reply:
It's been my assumption that most of the local work requires both the Specs repo and 1-All the SDK Language repos to be cloned locally. 

@praveenkuttappan do we have tools that are currently creating PRs for repos? Or is that done by out tool invoking a pipeline that has the PR creation logic?